src/TensorOneD.cpp

```diff
@@ -0,0 +1,140 @@
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+        : m_vec(other.m_vec)
+    {
+    }
+
+    TensorOneD::TensorOneD(TensorOneD &&other)
+        : m_vec(std::move(other.m_vec))
+    {
+    }
+
+    void TensorOneD::set_dim(std::size_t dim)
+    {
+        m_vec.resize(dim);
+    }
+
+    std::size_t TensorOneD::get_dim() const
+    {
+        return m_vec.size();
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        if (!input.is_array()) {
+            throw geopm::Exception("Neural network weights is non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().empty()) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        set_dim(input.array_items().size());
+        std::size_t vec_size = m_vec.size();
+
+        for (std::size_t idx = 0; idx < vec_size; ++idx) {
+            if (!input[idx].is_number()) {
+                throw geopm::Exception("Non-numeric type found in neural network weights.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_vec[idx] = input[idx].number_value();
+        }
+    }
+
+    TensorOneD TensorOneD::operator+(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Adding vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::plus<float>());
+
+        return rval;
+    }
+
+    TensorOneD TensorOneD::operator-(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Subtracting vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::minus<float>());
+
+        return rval;
+    }
+
+
+    float TensorOneD::operator*(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Inner product of vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        return std::inner_product(m_vec.begin(), m_vec.end(), other.m_vec.begin(), 0);
+    }
+
+    TensorOneD& TensorOneD::operator=(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+        return *this;
+    }
+
+    TensorOneD& TensorOneD::operator=(TensorOneD &&other)
+    {
+        if (&other == this) {
+            return *this;
+        }
+
+        m_vec = std::move(other.m_vec);
```

Comment:
Is there any particular reason why we want to use `std::move()`? Please correct me if I'm wrong, but doesn't `std::move()` transfer the information from the `other.m_vec` to the `m_vec`? Would the object/variable `other.m_vec` still be accessible after the `std::move()`? Does the `std::move()` modify the `other` object or it's internal data members in any way?

I guess what I'm trying to say is, what is the rationale for using
```cpp
m_vec = std::move(other.m_vec);
```
over using
```cpp
m_vec = other.m_vec;
```
as in the other `operator=()`?